CODINGSTYLE.md

```diff
@@ -204,6 +204,70 @@ In the following example, the empty string is the sentinel value indicating fail
 std::string parse_bss_info(const std::string &bss_info_str, db::bss_info_conf_t &bss_info_conf);
 ```
 
+## Comments
+Comments are important part of the code base.
+Comments are not compiled, nor have any influence on run time or any machinary.
+Comments are for humans, for developers, for us.
+Comments have the sole purpose of keeping the code maintainable.
+A good comment is a one that you, the writer of the comment, can understand after a while, after getting out of the context of writing it.
+Comments are written for future readers, it may be the future you, it may be someone else - 
+in this context in mind write your comments.
+Comments normaly explains the *why* and not the *what*.
+
+### Simple Comments
+It is useless to state the obvious:
+```cpp
+// checkig that the poinnter is not null
+if (ptr){
+}
+```
+
+However, it is extremely helpful to state our expectations
+```cpp
+// it is legal to receive an empty pointer.
+// in case of exiting pointer pointer we need to ...
+// and in case it is null we need to ...
+if (ptr){
+} else {
+}
+```
+
+### Spanning Comments
+There are cases that writing a comment accross few segments of the code, usually within a function's body, makes code understanding much easier.
+Sometimes it is hard to see behind the machincs of the code its intent.
+Adding descriptive comments that "tells the story" makes it much easier to follow the code structure and logic even before reading the code itself. Example:
```

Comment:
I really liked this part of your original comment, and it's missing here:

```
Sometimes the code is not trivial even if it doesn't do tricky things. 
It just has its logic that might be buried in a few lines of technical details that have nothing to do with the logic itself, like error exits.
In these cases it is more maintainable to comment the different sections in the function and their logic.
A small example can be seen [here](https://github.com/prplfoundation/prplMesh/blob/20388c47cf61f9419576fa223fd140e17adcb67d/agent/src/beerocks/slave/son_slave_thread.cpp#L4373-L4405).
This else clause starts with a story about what is done in it. 
In this case not what would you expect. 
It also states why it is fine and suggests future changes.
Later in the code the promise that was taken in the initial comment is implemented.
It is useful to read the first comment, then look forward in the code for more comments that support the story on top, and only then dive into the code to ensure it does what it says it does.
```

(I did minor editing of this text to make it fit with what you expect in documentation, e.g. avoid "I").

I think the code section that you refer to is a lot better than the example below, because it is real (and still simple enough).